savaslabs/savaslabs.github.io
# Issue #2146: Update to Bourbon 5

@oksana-c This PR does the following:

- Updates Bourbon to 5.0.0-beta.7, Neat to 2.0.0, and Bitters to 1.5.0. The updates to Bourbon and Neat were significant.
- Removes all old Bourbon mixins that were removed in version 5. The gist of version 5 is that many mixins that used to provide vendor prefixes were removed because developers are encouraged to use a post-processor with the autoprefixer module to add vendor prefixes. We are already doing this with gulp. Autoprefixer is in sync with the "Can I Use" website, which means it will be MUCH more up to date than the old Bourbon library ever was. An example of the updates I made here is that flexbox mixins like `display`, `justify-content` and `align-items` were removed and I am now able to write the plain CSS rules `display: flex`, `justify-content: center`, etc.
- Updates Neat mixins. The `span-columns` and `shift` mixins were renamed and slightly updated, so I've made these changes throughout the code.
- Refactors some of our base SCSS. The `_init.scss` file is no longer needed, and I broke the breakpoints out into their own file.
- Finally, some of my old posts about Bourbon are now outdated, so I added disclaimers to those posts. I opened an issue in the Content project to write a post about the update process so eventually I'll link to that post in the disclaimers of the outdated posts.

## To test

- Pull this code and run `gulp serve`. Make sure you don't get any error messages. You also should no longer see this error message:

```
autoprefixer: /Users/anne/Sites/savas-take-two/_assets/styles/main.css:4:36144: You should write display: flex by final spec instead of display: box
```

- Your local site should look 95% the same. There were a couple of regressions I caught but didn't bother fixing since we'll be overhauling the design:
  - The footer items aren't spaced properly
  - On small screens, there's too much left and right padding for things like blog cards, so the items appear to be too narrow
- Let me know if you see anything egregious, but I think we should avoid spending time fixing old styles since we'll be writing new ones!

Thanks! 🍊 🍋 :kiwi_fruit:
